glotaran/model/model.py

```diff
@@ -1,442 +1,476 @@
-            default_megacomplex_type=default_megacomplex_type,
-            dataset_group_models=dataset_group_models,
-        )
+        super().__init__(f"ModelError: {error}")
 
-        # iterate over items
-        for item_name, items in list(model_dict.items()):
 
-            if item_name not in model.model_items:
-                warn(f"Unknown model item type '{item_name}'.")
-                continue
+root_parameter_error = ModelError(
+    "The root parameter group cannot contain both groups and parameters."
+)
 
-            if isinstance(getattr(model, item_name), list):
-                model._add_list_items(item_name, items)
-            else:
-                model._add_dict_items(item_name, items)
-
-        return model
-
-    def _add_dict_items(self, item_name: str, items: dict):
-
-        for label, item in items.items():
-            item_cls = self.model_items[item_name]
-            if hasattr(item_cls, "_glotaran_model_item_typed"):
-                if "type" not in item and item_cls.get_default_type() is None:
-                    raise ValueError(f"Missing type for attribute '{item_name}'")
-                item_type = item.get("type", item_cls.get_default_type())
-
-                types = item_cls._glotaran_model_item_types
-                if item_type not in types:
-                    raise ValueError(f"Unknown type '{item_type}' for attribute '{item_name}'")
-                item_cls = types[item_type]
-            item["label"] = label
-            item = item_cls.from_dict(item)
-            getattr(self, item_name)[label] = item
-
-    def _add_list_items(self, item_name: str, items: list):
-
-        for item in items:
-            item_cls = self.model_items[item_name]
-            if hasattr(item_cls, "_glotaran_model_item_typed"):
-                if "type" not in item:
-                    raise ValueError(f"Missing type for attribute '{item_name}'")
-                item_type = item["type"]
-
-                if item_type not in item_cls._glotaran_model_item_types:
-                    raise ValueError(f"Unknown type '{item_type}' for attribute '{item_name}'")
-                item_cls = item_cls._glotaran_model_item_types[item_type]
-            item = item_cls.from_dict(item)
-            getattr(self, item_name).append(item)
-
-    def _add_megacomplexe_types(self):
-
-        for megacomplex_name, megacomplex_type in self._megacomplex_types.items():
-            if not issubclass(megacomplex_type, Megacomplex):
-                raise TypeError(
-                    f"Megacomplex type {megacomplex_name}({megacomplex_type}) "
-                    "is not a subclass of Megacomplex"
-                )
-            self._add_megacomplex_type(megacomplex_type)
 
-        model_megacomplex_type = create_model_megacomplex_type(
-            self._megacomplex_types, self.default_megacomplex
+def _load_item_from_dict(
+    item_type: type[Item], value: Item | Mapping, extra: dict[str, Any] = {}
```

Comment:
A mutable default arg is set at definition time and might lead to unexpected behavior, please make change the default of `extras` from `{}`  to `None` and set it to `{}` in the function body.